### PR TITLE
Add amber login support

### LINKED ISSUE #3: Log in with Amber

### PR DESCRIPTION
**Pull request by frnandu**
fixes #3

### CODE REVIEW

File: lib/widgets/app_drawer.dart [hunk 1]
```diff
@@ -107,6 +103,43 @@ class LoginContainer extends HookConsumerWidget {
...
+                  ? AsyncButtonBuilder(
+                      loadingWidget: SizedBox(
+                        width: 14,
+                        height: 14,
+                        child: CircularProgressIndicator(),
+                      ),
+                      onPressed: () async {
+                        if (await amber.isAppInstalled()) {
+                          final npub = await getExternalSignerPublicKey().catchError((e, stack) {
+                            print("error");
```
Comment (franzaps) [inline lib/widgets/app_drawer.dart:117]: Better to show a toast

File: lib/widgets/app_drawer.dart [hunk 2]
```diff
@@ -107,6 +103,43 @@ class LoginContainer extends HookConsumerWidget {
                   ),
                 ),
               ),
+              Gap(10),
+              isAndroid()
```
Comment (franzaps) [inline lib/widgets/app_drawer.dart:107]: Use `Platform.isAndroid` directly, it doesn't throw

File: lib/widgets/app_drawer.dart [hunk 3]
```diff
@@ -121,4 +154,31 @@ class LoginContainer extends HookConsumerWidget {
       ],
     );
   }
+
+  Future<void> doLogin(WidgetRef ref, String pubKey, BuildContext context) {
+    return ref.users.findOne(pubKey).then((user) {
+      ref.settings.findOneLocalById('_')!.user.value =
```
Comment (franzaps) [inline lib/widgets/app_drawer.dart:160]: We'd need to somehow store the fact that this logged in user can actually sign things

**Comment by frnandu** [comment]
>     * I was planning to use https://github.com/chebizarro/flutter-signer-plugin (vs Amberflutter) as it seems better and supports other NIP-55 signers. Also, what is the status of NIP-55 in NDK? Maybe it should be done through NDK?

We'll do it in NDK for sure, it's just not very high priority atm.
I just though we could already have amber support ASAP, since there are currently (AFAIK) no other signers besides Amber anyways.

>     * I want to keep the ability to sign in with pubkey or NIP-05
Hmm, should still work.

File: lib/widgets/app_drawer.dart (hunk 3 above)
Comment (franzaps) [reply lib/widgets/app_drawer.dart:160]: Exactly! If user "signed in" with pubkey we shouldn't even show options to sign things.

Let me think about it, I was meaning to revisit storing local user data in settings (`ref.settings.findOneLocalById('_')`), I don't really like the way it's currently done. As for the data structure I'd go with an enum as we'll support other signing methods in the future.

**Comment by franzaps** [comment]
Cool, no rush with NDK, and also no rush with this – we have nothing to sign at the moment (except zaps). As far as I understand NWC would require NDK, correct?

https://github.com/chebizarro/flutter-signer-plugin is more recent and appears higher quality than amberflutter, so I'd go with that.

> Hmm, should still work.

I haven't run the branch, but from what I see in the code logging in with external signer is the only option.

File: lib/widgets/app_drawer.dart (hunk 2 above)
Comment (franzaps) [reply lib/widgets/app_drawer.dart:107]: Actually, the check should not be for Android, it should be for `await amber.isAppInstalled()`. Do not show if it's not installed, do not launch a URL to download it

**Comment by frnandu** [comment]
> I haven't run the branch, but from what I see in the code logging in with external signer is the only option.

I didn't touch the existing login by npub/nip-05, neither the UI for that, so it should be possible to login in either by external signer or by old method.

**Comment by franzaps** [comment]
> > I haven't run the branch, but from what I see in the code logging in with external signer is the only option.
> 
> I didn't touch the existing login by npub/nip-05, neither the UI for that, so it should be possible to log in either by external signer or by old method.

You're right, my mistake

**Comment by franzaps** [comment]
Implemented based on this. Thanks!

**Pull request closed without merging**